spec/lang/well-formed.md

```diff
@@ -367,12 +367,12 @@ impl ValueExpr {
                         // If the pointer does not have metadata, this will still be well-formed but return the unit type.
                         ptr_ty.meta_kind().ty::<T>().unwrap_or_else(unit_type)
                     }
-                    SizeOfVal => {
-                        // TODO(UnsizedTypes): Support raw pointers as well.
-                        ensure_wf(
-                            matches!(operand, Type::Ptr(PtrType::Ref { .. })),
-                            "UnOp::SizeOfVal: invalid operand: not a reference"
-                        )?;
+                    ComputeSize(ty) | ComputeAlign(ty) => {
+                        // A thin pointer can also be the target type, with unit metadata.
+                        let meta_ty = ty.meta_kind().ty::<T>().unwrap_or_else(unit_type);
+                        if operand != meta_ty {
+                            throw_ill_formed!("UnOp::ComputeSize|ComputeAlign: invalid right type: not metadata of type");
```

Comment:
This is an un op, so "right" makes little sense I think? "invalid operand type" or something like that should work.